client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -475,9 +503,61 @@ function getQuickstartUrl( {
 	if ( orderId ) {
 		return;
 	}
+
+	// This is used when we need to quickly dial back the volume of concierge sessions
+	// being offered and sold, to be inline with HE availability.
+	// Change this percentage to change the amount of offers given out:
+	const percentConciergeOffers = 75;
+
+	if ( ! badNaiveClientSideRollout( 'conciergeUpsellDial', percentConciergeOffers ) ) {
+		return;
+	}
+
 	return `/checkout/offer-quickstart-session/${ pendingOrReceiptId }/${ siteSlug }`;
 }
 
+function getProfessionalEmailUpsellUrl( {
+	pendingOrReceiptId,
+	cart,
+	siteSlug,
+	orderId,
+	domains,
+}: {
+	pendingOrReceiptId: string;
+	cart: ResponseCart | undefined;
+	siteSlug: string | undefined;
+	orderId: number | undefined;
+	domains: Domain[] | undefined;
+} ): string | undefined {
+	if ( orderId || ! cart ) {
+		return;
+	}
+
+	if (
+		! config.isEnabled( 'upsell/professional-email' ) ||
+		hasGoogleApps( cart ) ||
+		hasTitanMail( cart )
+	) {
+		return;
+	}
+
+	if ( hasPremiumPlan( cart ) ) {
+		return;
+	}
+
+	if ( ! hasBloggerPlan( cart ) && ! hasPersonalPlan( cart ) && ! hasBusinessPlan( cart ) ) {
+		return;
+	}
+
+	const domain = getEligibleTitanDomain( siteSlug, domains );
```

Comment:
For this use case, I think we specifically want to offer the free trial, so it might be worth passing in an optional additionalFilter functional that is used to filter the list. Something like the following:
```suggestion
	const domain = getEligibleTitanDomain( siteSlug, domains, isDomainEligibleForTitanFreeTrial );
```
My thinking here is that we then _know_ that we're going to offer a free trial to the user.

I am also wondering if we need to be checking `cart` to see if it has a new domain registration using `hasNewDomainRegistration( cart )`, as that is probably the domain we should target if we can. I am honestly wondering if it makes more sense to just specify the domain name in the URL so we're only working all of this out once.